sbt updates should notify users they may be time consuming

jamesfe opened the pull request:
I am unsure if this is the right way to go about this - it seems like maybe the 'reason' variable may have some further information (i.e. "is this the first time sbt is getting run") but I am unsure where `apply` is called.  Either way, if sbt is being updated, it is likely that the dependencies will also be updated, taking quite some time.

There are a few things I am interested in reviewers commenting on: 
- Firstly, is this the right way to go about this?
- Second, should I have written tests?  It doesn't look like there are many (any?), so I would love to pair up with someone on that since I am new to scala.

Thanks in advance for your help -
James

eed3si9n commented on launcher-implementation/src/main/scala/xsbt/boot/Update.scala:
```diff
@@ -142,6 +142,9 @@ final class Update(config: UpdateConfiguration) {
           System.out.println("Getting " + app.groupID + " " + resolvedName + " " + app.getVersion + " " + reason + "...")
+          if (resolvedName == "sbt") {
+            System.out.println("Updating sbt, this may take some time.")
+          }
```
I don't know if we should hardcode this to sbt. Maybe you could just add it after `reason + ". "` or something.

eed3si9n commented:
Since this is about displaying messages on screen, I'm ok with not accompanying test, given that you can run it locally and see that it works.

jamesfe commented:
OK - I changed it so all updates will notify the user that they could take time.  There is another way to do this only for updates to `sbt`, I could create a string and append it to the println message in the event that it is `sbt` that is being updated, but that would still tie us to `sbt` and no other updates.

Also, I squashed the commits down.

jamesfe commented:
@eed3si9n @dwijnand Thanks for your help!
